operator/builtin/transformer/flatten/flatten_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package flatten
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/open-telemetry/opentelemetry-log-collection/entry"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
+	"github.com/open-telemetry/opentelemetry-log-collection/testutil"
+)
+
+type testCase struct {
+	name      string
+	expectErr bool
+	op        *FlattenOperatorConfig
+	input     func() *entry.Entry
+	output    func() *entry.Entry
+}
+
+// test building and processing a given config.
+func TestBuildAndProcess(t *testing.T) {
+	newTestEntry := func() *entry.Entry {
+		e := entry.New()
+		e.Timestamp = time.Unix(1586632809, 0)
+		e.Body = map[string]interface{}{
+			"key": "val",
+			"nested": map[string]interface{}{
+				"nestedkey": "nestedval",
+			},
+		}
+		return e
+	}
+	cases := []testCase{
+		{
+			"flatten_one_level",
+			false,
+			func() *FlattenOperatorConfig {
+				cfg := defaultCfg()
+				cfg.Field = entry.BodyField{
+					Keys: []string{"nested"},
+				}
+				return cfg
+			}(),
+			newTestEntry,
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Body = map[string]interface{}{
+					"key":       "val",
+					"nestedkey": "nestedval",
+				}
+				return e
+			},
+		},
+		{
+			"flatten_second_level",
+			false,
+			func() *FlattenOperatorConfig {
+				cfg := defaultCfg()
+				cfg.Field = entry.BodyField{
+					Keys: []string{"nested", "secondlevel"},
+				}
+				return cfg
+			}(),
+			func() *entry.Entry {
+				e := newTestEntry()
+				e.Body = map[string]interface{}{
+					"key": "val",
+					"nested": map[string]interface{}{
+						"secondlevel": map[string]interface{}{
+							"nestedkey": "nestedval",
+						},
+					},
+				}
```

Comment:
Let's have a test that uses this input but flattens to first level. I want to see that the deeper levels are still intact.